### PR TITLE
Added velocity checks to PushNotifications for experimental features

### PR DESCRIPTION
**Pull request by pmpurifoy**

### CODE REVIEW

File: build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml [hunk 1]
```diff
@@ -83,6 +83,14 @@ steps:
       filePath: tools\TerminalVelocity\Generate-TerminalVelocityFeatures.ps1
       arguments: -Path $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-DeploymentAPI.xml -Channel ${{ parameters.channel }} -Language C++ -Namespace Microsoft.Windows.ApplicationModel.WindowsAppSDK -Output $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-DeploymentAPI.h
       workingDirectory: '$(Build.SourcesDirectory)'
+
+  - task: powershell@2
+    displayName: 'Create DeploymentAPI TerminalVelocity features'
```
Comment (hulumane) [inline build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml:88]: Is the DisplayName accurate? Its PushNotificationAPI, not DeploymentAPI.

File: dev/Common/TerminalVelocityFeatures-PushNotifications.h [hunk 2]
```diff
@@ -3,21 +3,21 @@
...
+#define WINDOWSAPPRUNTIME_MICROSOFT_WINDOWS_PUSHNOTIFICATIONS_FEATURE_PUSHNOTIFICATIONS_ENABLED 1
 
 #if defined(__cplusplus)
 
 namespace Microsoft::Windows::PushNotifications
 {
 
-__pragma(detect_mismatch("ODR_violation_WINDOWSAPPRUNTIME_MICROSOFT_WINDOWS_PUSHNOTIFICATIONS_FEATURE_DEPLOYMENTAPI_ENABLED_mismatch", "AlwaysEnabled"))
-struct Feature_DeploymentAPI
+__pragma(detect_mismatch("ODR_violation_WINDOWSAPPRUNTIME_MICROSOFT_WINDOWS_PUSHNOTIFICATIONS_FEATURE_PUSHNOTIFICATIONS_ENABLED_mismatch", "AlwaysEnabled"))
```
Comment (hulumane) [inline dev/Common/TerminalVelocityFeatures-PushNotifications.h:17]: Does this mean that by default, our experimental APIs are going to just work here? If so, any reason for having a velocity check in the first place?

**Review by hulumane — APPROVED** [review]

File: dev/Common/TerminalVelocityFeatures-PushNotifications.h (hunk 2 above)
Comment (pmpurifoy) [reply dev/Common/TerminalVelocityFeatures-PushNotifications.h:17]: In the pipeline, a script "Generate-TerminalVelocityFeatures.cmd" will create this header file and the constant WINDOWSAPPRUNTIME_MICROSOFT_WINDOWS_PUSHNOTIFICATIONS_FEATURE_PUSHNOTIFICATIONS_ENABLED, will be set depending on the -Channel flag passed. This .h was created with the 'Experimental' flag but will be overrode in the pipeline with 'Preview/Stable' flags where a different header file will be made.

Here is a link to some comments from @DrusTheAxe  about it:
https://github.com/microsoft/WindowsAppSDK/pull/1403#discussion_r707576162

**Review by hulumane — APPROVED** [review]

File: build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml [hunk 3]
```diff
@@ -83,6 +83,14 @@ steps:
       filePath: tools\TerminalVelocity\Generate-TerminalVelocityFeatures.ps1
       arguments: -Path $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-DeploymentAPI.xml -Channel ${{ parameters.channel }} -Language C++ -Namespace Microsoft.Windows.ApplicationModel.WindowsAppSDK -Output $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-DeploymentAPI.h
       workingDirectory: '$(Build.SourcesDirectory)'
+
+  - task: powershell@2
+    displayName: 'Create PushNotifications TerminalVelocity features'
+    inputs:
+      targetType: filePath
+      filePath: tools\TerminalVelocity\Generate-TerminalVelocityFeatures.ps1
+      arguments: -Path $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-PushNotifications.xml -Channel ${{ parameters.channel }} -Language C++ -Namespace Microsoft.Windows.PushNotifications -Output $(Build.SourcesDirectory)\dev\common\TerminalVelocityFeatures-PushNotifications.h
```
Comment (danielayala94) [inline build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml:92]: Is parameters.channel = AlwaysEnabled, DisabledByDefault and such kind of values?

Comment (danielayala94) [inline build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml:92]: What kind of values can `parameters.channel` take in?

**Review by danielayala94 — APPROVED** [review]

File: build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml (hunk 3 above)
Comment (pmpurifoy) [reply build/AzurePipelinesTemplates/WindowsAppSDK-BuildProject-Steps.yml:92]: Preview, Stable, Experimental. Check out the docs located in docs/CodingGuidelines/TerminalVelocity.md for more information on the other flags used in the script 😃

**Review by DrusTheAxe — APPROVED** [review]

**Pull request merged**